libs/ledger-live-common/src/deviceSDK/hooks/useToggleOnboardingEarlyChecks.ts

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from "react";
+import { log } from "@ledgerhq/logs";
+import {
+  ToggleOnboardingEarlyCheckActionArgs,
+  ToggleOnboardingEarlyCheckActionState,
+  toggleOnboardingEarlyCheckAction as defaultToggleOnboardingEarlyCheckAction,
+  initialState,
+} from "../actions/toggleOnboardingEarlyCheck";
+
+export type UseToggleOnboardingEarlyCheckArgs = {
+  toggleOnboardingEarlyCheckAction?: typeof defaultToggleOnboardingEarlyCheckAction;
+  deviceId: string;
+  toggleType: ToggleOnboardingEarlyCheckActionArgs["toggleType"] | null;
+};
+
+/**
+ * Hook making the device enter or exit the early security check steps, during the onboarding
+ *
+ * This hook only puts (or moves out) the device to the state/step of the early security check.
+ * It does not starts any "security checks".
+ *
+ * If the device is not in the WELCOME or WELCOME_STEP2 onboarding state, this hook will update
+ * its `toggleStatus` to `failure`.
+ *
+ * You can reset the result state by updating `toggleType` to null
+ *
+ * @param deviceId The id of the targeted device that can be
+ * @param toggleType either null, "enter" or "exit".
+ *  If null, the hook does not subscribe to the underlying device action, and nothing happens.
+ *  Useful to enable/disable this hook.
+ * @param toggleOnboardingEarlyCheckAction dependency injected action. A default implementation is provided.
+ * @returns an object containing the state of the onboarding early check toggling with possible error.
+ *   The resulting state is reset to its initial state on each new triggering of the toggle action (when `toggleType` is updated):
+ * {
+ *   toggleStatus: "none",
+ *   ...initialSharedActionState,
+ * }
+ */
+export const useToggleOnboardingEarlyCheck = ({
```

Comment:
With this approach, the command is only sent to the device (for a fixed device and implementation) if the `"toggleType"` changes, is that something that is ok? 
For instance consider this scenario:
- in my component I want to use this command once by setting `enter` in an internal state and passing that state to this hook
- the command fails and the hook returns a "failure" status (for instance because the device wasn't in the correct state)
- what can I do to re-trigger the command if I want ?

2 solutions I see:
- Could there be at least a nonce parameter or something to "force re-trigger" the command?
- Another approach for this hook would be: instead of triggering the command based on a state, just make this hook return a function that does the triggering. Then it's up to the consumer to implement the way they want to trigger that command (they can use an internal state and a useEffect in a similar fashion if they want, they can also just use the function returned by the hook and call it whenever they want)

But maybe the scenario above is not likely to happen in our use cases so we don't care 🤷‍♂️

Reply:
actually I can recover by setting to `null` and then back to the same value so maybe it's fine like that